### PR TITLE
Move contracts/FixedFeeRegistrar to semanticTests/externalContracts

### PR DESCRIPTION
**Pull request by axic**
Depends on #11423.

### CODE REVIEW

File: test/libsolidity/semanticTests/externalContracts/FixedFeeRegistrar.sol [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+// reserve(string), 69 ether: 0x20, 3, "abc" ->
+// gas irOptimized: 45527
+// gas legacy: 46242
+// gas legacyOptimized: 45491
+// owner(string): 0x20, 3, "abc" -> 0x1212121212121212121212121212120000000012
+// reserve(string), 70 ether: 0x20, 3, "def" ->
+// owner(string): 0x20, 3, "def" -> 0x1212121212121212121212121212120000000012
+// reserve(string), 68 ether: 0x20, 3, "ghi" ->
+// owner(string): 0x20, 3, "ghi" -> 0
+// account: 1 -> 0x1212121212121212121212121212120000001012
```
Comment (axic) [inline test/libsolidity/semanticTests/externalContracts/FixedFeeRegistrar.sol:91]: This is not 100% complete yet, such as the properties and disown test is missing, however without having comments/separation it is quite hard to follow, so would need #11427.

**Comment by axic** [comment]
I think this is good enough, though haven't added all cases form the `properties` loop.

File: test/libsolidity/semanticTests/externalContracts/FixedFeeRegistrar.sol [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+// content(string): 0x20, 3, "abc" -> 0x00
+// setContent(string,bytes32): 0x40, 333, 3, "def" ->
+// setAddr(string,address): 0x40, 124, 3, "def" ->
+// setSubRegistrar(string,address): 0x40, 125, 3, "def" ->
+// content(string): 0x20, 3, "def" -> 333
+// addr(string): 0x20, 3, "def" -> 124
+// subRegistrar(string): 0x20, 3, "def" -> 125
+// balance: 0x124 -> 0
+// disown(string,address): 0x40, 0x124, 3, "def" ->
+// balance: 0x124 -> 0
```
Comment (cameel) [inline test/libsolidity/semanticTests/externalContracts/FixedFeeRegistrar.sol:107]: Looks like you did not include the `"reserve(string)", m_fee` part from the `disown` test so this is zero here rather than `m_fee` (i.e. 69 ETH) as in the original test. Is that an intentional change?

File: test/libsolidity/semanticTests/externalContracts/FixedFeeRegistrar.sol (hunk 1 above)
Comment (cameel) [reply test/libsolidity/semanticTests/externalContracts/FixedFeeRegistrar.sol:91]: Maybe you could at least use the `#` comments to mark where tests from the original start/end?

As far as I can see, you translated `creation`, `reserve`, `transfer` and `disown` (the latter two with slight modifications) and dropped `double_reserve` and `properties`?

**Review by cameel — COMMENTED** [review]
The part that has been translated looks good. Will you be adding the rest of the tests later or do you think they're not necessary?

File: test/libsolidity/semanticTests/externalContracts/FixedFeeRegistrar.sol (hunk 2 above)
Comment (axic) [reply test/libsolidity/semanticTests/externalContracts/FixedFeeRegistrar.sol:107]: Missed because it is hard to read due to the lack of spaces/comments.

File: test/libsolidity/semanticTests/externalContracts/FixedFeeRegistrar.sol (hunk 1 above)
Comment (axic) [reply test/libsolidity/semanticTests/externalContracts/FixedFeeRegistrar.sol:91]: Only properties is missing. Everything else is there. I tried adding the `#` comments but they only work in specific parts and wherever I put that I got a parsing error.

Comment (cameel) [reply test/libsolidity/semanticTests/externalContracts/FixedFeeRegistrar.sol:91]: `double_reserve` seems missing too.

Well, there's this part but if that's it then it's just half of it:
```solidity
// reserve(string), 69 ether: 0x20, 3, "abc" ->
// owner(string): 0x20, 3, "abc" -> 0x1212121212121212121212121212120000000012
// account: 0 -> 0x1212121212121212121212121212120000000012
```
But I see that the other half requires sending ether form another account. I guess that's not currently possible in these tests?

Comment (axic) [reply test/libsolidity/semanticTests/externalContracts/FixedFeeRegistrar.sol:91]: `double_reserve` is not missing, just that the C++ version *reset* the state every single test case and we don't.

**Review by cameel — APPROVED** [review]
There are still some minor issues but I'm approving anyway because I think it's good enough overall and maybe these details are not worth spending time on.

**Comment by cameel** [comment]
Looks like this PR needed a rebase and regenerating test output before merge. Tests are failing for me on develop:
```
/solidity/test/boostTest.cpp(127): error: in "semanticTests/externalContracts/FixedFeeRegistrar": Test expectation mismatch.
Expected result:
// constructor()
// gas irOptimized: 544916
// gas legacy: 942808
// gas legacyOptimized: 492291
// reserve(string), 69 ether: 0x20, 3, "abc" ->
// gas irOptimized: 45527
// gas legacy: 46242
// gas legacyOptimized: 45491
// owner(string): 0x20, 3, "abc" -> 0x1212121212121212121212121212120000000012
// reserve(string), 70 ether: 0x20, 3, "def" ->
// owner(string): 0x20, 3, "def" -> 0x1212121212121212121212121212120000000012
// reserve(string), 68 ether: 0x20, 3, "ghi" ->
// owner(string): 0x20, 3, "ghi" -> 0
// account: 1 -> 0x1212121212121212121212121212120000001012
// reserve(string), 69 ether: 0x20, 3, "abc" ->
// owner(string): 0x20, 3, "abc" -> 0x1212121212121212121212121212120000000012
// account: 0 -> 0x1212121212121212121212121212120000000012
// setContent(string,bytes32): 0x40, 0, 3, "abc" ->
// transfer(string,address): 0x40, 555, 3, "abc" ->
// owner(string): 0x20, 3, "abc" -> 555
// content(string): 0x20, 3, "abc" -> 0x00
// setContent(string,bytes32): 0x40, 333, 3, "def" ->
// setAddr(string,address): 0x40, 124, 3, "def" ->
// setSubRegistrar(string,address): 0x40, 125, 3, "def" ->
// content(string): 0x20, 3, "def" -> 333
// addr(string): 0x20, 3, "def" -> 124
// subRegistrar(string): 0x20, 3, "def" -> 125
// balance: 0x124 -> 0
// disown(string,address): 0x40, 0x124, 3, "def" ->
// balance: 0x124 -> 0
// owner(string): 0x20, 3, "def" -> 0
// content(string): 0x20, 3, "def" -> 0
// addr(string): 0x20, 3, "def" -> 0
// subRegistrar(string): 0x20, 3, "def" -> 0

Obtained result:
// constructor()
// gas irOptimized: 544916
// gas legacy: 942808
// gas legacyOptimized: 492291
// reserve(string), 69 ether: 0x20, 3, "abc" ->
// ~ emit Changed(string): #0x4e03657aea45a94fc7d47ba826c8d667c0d1e6e33a64a036ec44f58fa12d6c45
// gas irOptimized: 45527
// gas legacy: 46242
// gas legacyOptimized: 45491
// owner(string): 0x20, 3, "abc" -> 0x1212121212121212121212121212120000000012
// reserve(string), 70 ether: 0x20, 3, "def" ->
// ~ emit Changed(string): #0x34607c9bbfeb9c23509680f04363f298fdb0b5f9abe327304ecd1daca08cda9c
// owner(string): 0x20, 3, "def" -> 0x1212121212121212121212121212120000000012
// reserve(string), 68 ether: 0x20, 3, "ghi" ->
// owner(string): 0x20, 3, "ghi" -> 0
// account: 1 -> 0x1212121212121212121212121212120000001012
// reserve(string), 69 ether: 0x20, 3, "abc" ->
// owner(string): 0x20, 3, "abc" -> 0x1212121212121212121212121212120000000012
// account: 0 -> 0x1212121212121212121212121212120000000012
// setContent(string,bytes32): 0x40, 0, 3, "abc" ->
// ~ emit Changed(string): #0x4e03657aea45a94fc7d47ba826c8d667c0d1e6e33a64a036ec44f58fa12d6c45
// transfer(string,address): 0x40, 555, 3, "abc" ->
// ~ emit Changed(string): #0x4e03657aea45a94fc7d47ba826c8d667c0d1e6e33a64a036ec44f58fa12d6c45
// owner(string): 0x20, 3, "abc" -> 555
// content(string): 0x20, 3, "abc" -> 0x00
// setContent(string,bytes32): 0x40, 333, 3, "def" ->
// ~ emit Changed(string): #0x34607c9bbfeb9c23509680f04363f298fdb0b5f9abe327304ecd1daca08cda9c
// setAddr(string,address): 0x40, 124, 3, "def" ->
// ~ emit Changed(string): #0x34607c9bbfeb9c23509680f04363f298fdb0b5f9abe327304ecd1daca08cda9c
// setSubRegistrar(string,address): 0x40, 125, 3, "def" ->
// ~ emit Changed(string): #0x34607c9bbfeb9c23509680f04363f298fdb0b5f9abe327304ecd1daca08cda9c
// content(string): 0x20, 3, "def" -> 333
// addr(string): 0x20, 3, "def" -> 124
// subRegistrar(string): 0x20, 3, "def" -> 125
// balance: 0x124 -> 0
// disown(string,address): 0x40, 0x124, 3, "def" ->
// ~ emit Changed(string): #0x34607c9bbfeb9c23509680f04363f298fdb0b5f9abe327304ecd1daca08cda9c
// balance: 0x124 -> 0
// ~ emit Changed(string): #0x34607c9bbfeb9c23509680f04363f298fdb0b5f9abe327304ecd1daca08cda9c
// owner(string): 0x20, 3, "def" -> 0
// content(string): 0x20, 3, "def" -> 0
// addr(string): 0x20, 3, "def" -> 0
// subRegistrar(string): 0x20, 3, "def" -> 0
```
